WeScanTests/UIImageTests.swift

```diff
@@ -11,69 +11,85 @@ import FBSnapshotTestCase
-    view.sizeToFit()
+    func testRotateDefaultFacingImageCorrectly() {
+        let image = UIImage(named: ResourceImage.rect2.rawValue, in: Bundle(for: ImageFeatureTestHelpers.self), compatibleWith: nil)
+        let orientatedImage = UIImage(cgImage: image!.cgImage!, scale: 1.0, orientation: .rightMirrored)
+        
+        let view = UIImageView(image: orientatedImage.applyingPortraitOrientation())
+        view.sizeToFit()
+        
+        FBSnapshotVerifyView(view)
+    }
     
-    FBSnapshotVerifyView(view)
-  }
-  
-  func testRotateRightFacingImageCorrectly() {
-    let image = UIImage(named: ResourceImage.rect2.rawValue, in: Bundle(for: ImageFeatureTestHelpers.self), compatibleWith: nil)
-    let orientatedImage = UIImage(cgImage: image!.cgImage!, scale: 1.0, orientation: .right)
+    func testRotateImageCorrectly() {
+        let image = UIImage(named: ResourceImage.rect2.rawValue, in: Bundle(for: ImageFeatureTestHelpers.self), compatibleWith: nil)
+        
+        let view = UIImageView(image: image!.rotated(by: Measurement(value: Double.pi * 0.2, unit: .radians), options: []))
+        view.sizeToFit()
+        
+        FBSnapshotVerifyView(view)
+    }
     
-    let view = UIImageView(image: orientatedImage.applyingPortraitOrientation())
-    view.sizeToFit()
+    func testScaledImageSuccessfully() {
+        let image = UIImage(named: ResourceImage.rect2.rawValue, in: Bundle(for: ImageFeatureTestHelpers.self), compatibleWith: nil)!
+        XCTAssertNotNil(image.scaledImage(scaleFactor: 0.2))
+    }
     
-    FBSnapshotVerifyView(view)
-  }
-  
-  func testRotateDefaultFacingImageCorrectly() {
-    let image = UIImage(named: ResourceImage.rect2.rawValue, in: Bundle(for: ImageFeatureTestHelpers.self), compatibleWith: nil)
-    let orientatedImage = UIImage(cgImage: image!.cgImage!, scale: 1.0, orientation: .rightMirrored)
+    func testScaledImageCorrectly() {
+        let image = UIImage(named: ResourceImage.rect2.rawValue, in: Bundle(for: ImageFeatureTestHelpers.self), compatibleWith: nil)!
+        XCTAssertEqual(image.size, CGSize(width: 500, height: 500))
+        XCTAssertEqual(image.scaledImage(scaleFactor: 0.2)!.size, CGSize(width: 100, height: 100))
+    }
     
-    let view = UIImageView(image: orientatedImage.applyingPortraitOrientation())
-    view.sizeToFit()
-    
-    FBSnapshotVerifyView(view)
-  }
-  
-  func testRotateImageCorrectly() {
-    let image = UIImage(named: ResourceImage.rect2.rawValue, in: Bundle(for: ImageFeatureTestHelpers.self), compatibleWith: nil)
-    
-    let view = UIImageView(image: image!.rotated(by: Measurement(value: Double.pi * 0.2, unit: .radians), options: []))
-    view.sizeToFit()
-    
-    FBSnapshotVerifyView(view)
-  }
+    func testPDFDataCreationSuccessful() {
+        let image = UIImage(named: ResourceImage.rect2.rawValue, in: Bundle(for: ImageFeatureTestHelpers.self), compatibleWith: nil)!
+        XCTAssertNotNil(image.pdfData())
```

Comment:
Can we verify the PDF creation by saving an example scan data to `String` and match that here?

Reply:
I considered that - but the PDF renderer is really on Apple's side and I fear that if their implementation changes even a bit in the future it would affect the test?

Reply:
You might be right there! Good point. Let's keep it like this for now

Reply:
👍